core-api/src/main/java/com/optimizely/ab/notification/NotificationCenter.java

```diff
@@ -194,6 +204,13 @@ public int addNotificationListener(NotificationType notificationType, Notificati
      * @return true if removed otherwise false (if the notification is already registered, it returns false).
      */
     public boolean removeNotificationListener(int notificationID) {
+        for (NotificationHolder holder : decisionListenerHolder) {
```

Comment:
@mikecdavis What do you suggest here? There is separate logic to remove for DecisionListener object.